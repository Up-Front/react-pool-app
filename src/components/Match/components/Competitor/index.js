import React from 'react';
import PropTypes from 'prop-types';
import Constant from './../../../shared/constants';
import {
  CompetitorWrapper,
  CompetitorAvatar,
  CompetitorName,
  CompetitorLine,
} from './styles';

const Competitor = ({
  competitor,
  checkAuthIsCompetitor,
  hasVote,
  winner,
  ...props
}) => {
  const isWinner = winner === competitor.uid ? true : false;

  const handleClick = () => {
    if (checkAuthIsCompetitor && !winner) {
      props.handleClick(competitor);
    }
  };

  const authVote = () => hasVote && !winner;

  if (isWinner) {
    return <strong>{competitor.displayName} ðŸ‘‘</strong>;
  }
  return (
    <CompetitorWrapper onClick={handleClick} hasVote={authVote()}>
      <CompetitorLine align={props.align}>
        <CompetitorAvatar src={competitor.avatarUrl} />
        <CompetitorName
          className="name"
          align={props.align}
          hasVote={authVote()}
        >
          {competitor.displayName}
        </CompetitorName>
      </CompetitorLine>
    </CompetitorWrapper>
  );
};

Competitor.propTypes = {
  align: PropTypes.string.isRequired,
  checkAuthIsCompetitor: PropTypes.bool,
  competitor: PropTypes.object.isRequired,
  hasVote: PropTypes.bool,
  winner: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};
export default Competitor;
