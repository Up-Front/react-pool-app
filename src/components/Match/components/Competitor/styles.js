import styled from 'styled-components';
import constants from './../../../shared/constants';

export const CompetitorWrapper = styled.div`
  flex: 1;
`;

export const CompetitorLine = styled.div`
  display: flex;
  flex-direction: ${props =>
    props.align === constants.ALIGNRIGHT ? 'row-reverse' : 'row'};
  align-items: center;
  width: 100%;
`;

export const CompetitorAvatar = styled.img`
  margin: 5px;
  width: 40px;
  height: 40px;
  border-radius: 50%;
`;

export const CompetitorName = styled.div`
  flex: 1;
  color: ${props => (props.hasVote ? 'green' : '')};
  text-align: ${props =>
    props.align === constants.ALIGNRIGHT ? 'right' : 'left'};
`;
