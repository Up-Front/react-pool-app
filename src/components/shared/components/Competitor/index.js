import React from 'react';
import PropTypes from 'prop-types';
import constants from './../../../shared/constants';
import RankingDetail from './../RankingDetail';
import {
  CompetitorWrapper,
  CompetitorAvatar,
  CompetitorName,
  CompetitorLine,
} from './styles';

const Competitor = ({
  competitor,
  checkAuthIsCompetitor,
  hasVote,
  ...props
}) => {
  const handleClick = () => {
    if (checkAuthIsCompetitor) {
      props.handleClick(competitor);
    }
  };

  const authVote = () => hasVote;

  return (
    <CompetitorWrapper onClick={handleClick} hasVote={authVote()}>
      <CompetitorLine align={props.align}>
        <CompetitorAvatar
          src={competitor.avatarUrl}
          online={competitor.online}
        />
        <RankingDetail
          eloRating={competitor.ranking}
          currentRanking={competitor.currentRanking}
          previousRanking={competitor.previousRanking}
        />
        <CompetitorName
          className="name"
          align={props.align}
          hasVote={authVote()}
        >
          {competitor.displayName}
        </CompetitorName>
      </CompetitorLine>
    </CompetitorWrapper>
  );
};

Competitor.defaultProps = {
  handleClick: () => {},
  online: false,
  align: constants.ALIGNLEFT,
};

Competitor.propTypes = {
  align: PropTypes.string.isRequired,
  checkAuthIsCompetitor: PropTypes.bool,
  competitor: PropTypes.object.isRequired,
  hasVote: PropTypes.bool,
  winner: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  online: PropTypes.bool,
};
export default Competitor;
