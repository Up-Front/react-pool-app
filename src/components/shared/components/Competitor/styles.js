import styled from 'styled-components';
import constants from './../../../shared/constants';

export const CompetitorWrapper = styled.div`
  flex: 1;
`;

export const CompetitorLine = styled.div`
  display: flex;
  flex-direction: ${props =>
    props.align === constants.ALIGNRIGHT ? 'row-reverse' : 'row'};
  align-items: center;
  width: 100%;
`;

export const CompetitorName = styled.div`
  flex: 1;
  color: ${props => (props.hasVote ? 'green' : '')};
  text-align: ${props =>
    props.align === constants.ALIGNRIGHT ? 'right' : 'left'};
`;
