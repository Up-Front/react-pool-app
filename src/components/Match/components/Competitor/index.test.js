import React from 'react';
import { shallow, mount } from 'enzyme';
import Constants from './../../../shared/constants';
import Competitor from './';

describe('Competitor component', () => {
  let competitor;
  let handleClick;
  beforeEach(() => {
    competitor = {
      uid: '1',
      displayName: 'Bob de Bouwer',
    };

    handleClick = jest.fn();
  });

  test('renders without crashing', () => {
    const wrapper = shallow(
      <Competitor
        align={Constants.ALIGNLEFT}
        competitor={competitor}
        handleClick={handleClick}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('handleClick', () => {
    const wrapper = shallow(
      <Competitor
        checkAuthIsCompetitor={true}
        competitor={competitor}
        handleClick={handleClick}
        align={Constants.ALIGNLEFT}
      />
    );
    wrapper.find('[onClick]').simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('check that you voted', () => {
    const wrapper = mount(
      <Competitor
        align={Constants.ALIGNLEFT}
        hasVote={true}
        competitor={competitor}
        handleClick={handleClick}
      />
    );
    expect(wrapper.find('div.name').length).toEqual(1);
  });

  test('render of winner', () => {
    const wrapper = mount(
      <Competitor
        winner="1"
        align={Constants.ALIGNLEFT}
        competitor={competitor}
        handleClick={handleClick}
      />
    );

    expect(wrapper.find('strong').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
