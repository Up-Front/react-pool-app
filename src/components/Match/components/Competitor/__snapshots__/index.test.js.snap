// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competitor component render of winner 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Competitor
    align="left"
    competitor={
        Object {
            "displayName": "Bob de Bouwer",
            "uid": "1",
          }
    }
    handleClick={[Function]}
    winner="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "align": "left",
      "competitor": Object {
        "displayName": "Bob de Bouwer",
        "uid": "1",
      },
      "handleClick": [Function],
      "winner": "1",
    },
    "ref": null,
    "rendered": Object {
      "instance": <strong>
        Bob de Bouwer
        ðŸ‘‘
</strong>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Bob de Bouwer",
          " ðŸ‘‘",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Bob de Bouwer",
        " ðŸ‘‘",
      ],
      "type": "strong",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "align": "left",
        "competitor": Object {
          "displayName": "Bob de Bouwer",
          "uid": "1",
        },
        "handleClick": [Function],
        "winner": "1",
      },
      "ref": null,
      "rendered": Object {
        "instance": <strong>
          Bob de Bouwer
          ðŸ‘‘
</strong>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bob de Bouwer",
            " ðŸ‘‘",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Bob de Bouwer",
          " ðŸ‘‘",
        ],
        "type": "strong",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Competitor component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Competitor
    align="left"
    competitor={
        Object {
            "displayName": "Bob de Bouwer",
            "uid": "1",
          }
    }
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        align="left"
>
        <styled.img
                src={undefined}
        />
        <styled.div
                align="left"
                className="name"
                hasVote={undefined}
        >
                Bob de Bouwer
        </styled.div>
</styled.div>,
      "hasVote": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "left",
        "children": Array [
          <styled.img
            src={undefined}
/>,
          <styled.div
            align="left"
            className="name"
            hasVote={undefined}
>
            Bob de Bouwer
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "left",
            "children": "Bob de Bouwer",
            "className": "name",
            "hasVote": undefined,
          },
          "ref": null,
          "rendered": "Bob de Bouwer",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          align="left"
>
          <styled.img
                    src={undefined}
          />
          <styled.div
                    align="left"
                    className="name"
                    hasVote={undefined}
          >
                    Bob de Bouwer
          </styled.div>
</styled.div>,
        "hasVote": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "left",
          "children": Array [
            <styled.img
              src={undefined}
/>,
            <styled.div
              align="left"
              className="name"
              hasVote={undefined}
>
              Bob de Bouwer
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "left",
              "children": "Bob de Bouwer",
              "className": "name",
              "hasVote": undefined,
            },
            "ref": null,
            "rendered": "Bob de Bouwer",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
